awot/io/read_t28.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import absolute_import, print_function
+import numpy as np
+from netCDF4 import num2date
+from .flight import _winduv
+
+
+def read_t28_netcdf(ncFile):
+    """
+    T-28 has enough differences from other aircraft netCDF files to
+    warrant its own read script.
+
+    Key differences are the large number of non-standard variables
+    (e.g., from the electric field meters) and how time is handled by
+    in the netCDF file.
+    """
+    name_map = _t28_flight_namemap()
+    data = _make_data_dictionary(ncFile, name_map)
+    data['time'] = _get_time(ncFile)
+    if 'turb' in data.keys():
+        CM2M_TWOTHIRDS = 100**0.6666667
+        data['turb'] /= CM2M_TWOTHIRDS
+
+    if 'Uwind' not in name_map:
+        Uwind, Vwind = _winduv(data)
+        # Add to the dictionary
+        data['Uwind'] = Uwind
+        data['Vwind'] = Vwind
+
+    data['project'] = ''
+    data['platform'] = 'SDSM&T T-28'
+    if hasattr(ncFile, 'FlightNumber'):
+        data['flight_number'] = ncFile.FlightNumber
+    else:
+        data['flight_number'] = ''
+
+    return data
+
+
+def _make_data_dictionary(ncFile, name_map):
+    """
+    T-28 provides 2D arrays (20-Hz sampling reporting every second),
+    and these need to be flattened.
+    """
+    data = {}
+    for var in name_map:
+        tmp = np.array(ncFile.variables[name_map[var]]).ravel()
+        data[var] = tmp.ravel()
+        try:
+            np.ma.masked_invalid(data[var])
+        except:
+            data[var] = None
+        try:
+            mask = data[var].mask
+        except:
+            data[var] = np.ma.masked_array(data[var], mask=False)
+    return data
+
+
+def _get_time(ncFile):
+    TimeSec = np.array(ncFile.variables['Time'][:]).ravel()
+    units = ncFile.variables['Time'].units
+    return num2date(TimeSec, units)
+
```

Comment:
Interesting, I didn't think to just make the TimeSec a Numpy array here. I ended up just creating a new array. I wonder if your method is a little faster. With high data rates this can take a small amount of time.